### ISSUE TITLE
Change Name getter in HtmlDocumentNavigator to return lower-case variant of name for (X)HTML element nodes

### ISSUE THREAD

**Issue by martin-honnen** [AngleSharp/AngleSharp.XPath#27]
Given that https://github.com/AngleSharp/AngleSharp.XPath/blob/master/src/AngleSharp.XPath/HtmlDocumentNavigator.cs#L57 simply does

```
        public override string Name =>
            _attrIndex != -1
                ? NameTable.GetOrAdd(CurrentElement.Attributes[_attrIndex].Name)
                : NameTable.GetOrAdd(_currentNode.NodeName);

```

and NodeName in the DOM for elements in the XHTML namespace https://dom.spec.whatwg.org/#element-html-uppercased-qualified-name uses the upper-cased name any XPath use on an HTML element (e.g. `p` element) will have `name()` give an upper-case name e.g. `P` while `local-name()` contradicts that as it gives a lower-case `p`. 

That case difference in the DOM world is contrary to what you have in normal XPath as there local-name() and name() will always have the same case for the letters of the name.

Therefore I think, to have a consistent XPath(Navigator) API, the `HtmlDocumentNavigator`  in its `Name` getter needs to ensure it lower-cases the names of HTML elements.

**Comment by martin-honnen** [comment]
Test showing the problem would be

```
        [Test]
        public void TestSimpleHTMLDocumentXPath()
        {
            string html = @"<html xmlns='http://www.w3.org/1999/xhtml'><head><title>Test</title></head><body><h1>Test</h1></body></html>";

            XmlDocument xmlDoc = new XmlDocument();
            xmlDoc.LoadXml(html);

            var angleSharpXmlDoc = new HtmlParser().ParseDocument(html);

            TestElementsOnXPathLevel(angleSharpXmlDoc.DocumentElement.SelectNodes("//*", false), xmlDoc.SelectNodes("//*"));

        }

```

where the function `TestElementsOnXPathLevel` does

```
        public void TestElementsOnXPathLevel(List<INode> angleSharpElements, XmlNodeList xmlElements)
        {
            foreach (var elementPair in angleSharpElements.OfType<IElement>().Zip(xmlElements.OfType<XmlElement>()))
            {
                var angleSharpNav = new HtmlDocumentNavigator(elementPair.First.Owner, elementPair.First, false);
                var xmlNav = elementPair.Second.CreateNavigator();

                Assert.AreEqual(angleSharpNav.Evaluate("namespace-uri()"), xmlNav.Evaluate("namespace-uri()"));
                Assert.AreEqual(angleSharpNav.Evaluate("local-name()"), xmlNav.Evaluate("local-name()"));
                Assert.AreEqual(angleSharpNav.Evaluate("name()"), xmlNav.Evaluate("name()"));
 
            }
        }

```

**Comment by martin-honnen** [comment]
Two simpler test are
```
       [Test]
        public void TestNameXPathFunctionOnXMLDoc()
        {
            string xml = @"<html><head><title>Test</title></head><body><h1>Test</h1></body></html>";

            var angleSharpXmlDoc = new XmlParser().ParseDocument(xml);

            var xmlNav = angleSharpXmlDoc.CreateNavigator();

            Assert.AreEqual(xmlNav.Evaluate("name()"), "html");

        }

        [Test]
        public void TestNameXPathFunctionOnHTMLDoc()
        {
            string html = @"<html><head><title>Test</title></head><body><h1>Test</h1></body></html>";

            var angleSharpHtmlDoc = new HtmlParser().ParseDocument(html);

            var htmlNav = angleSharpHtmlDoc.CreateNavigator();

            Assert.AreEqual(htmlNav.Evaluate("name()"), "html");
        }

```

`TestNameXPathFunctionOnHTMLDoc` fails with current release of AngleSharp.XPath.

**Comment by denis-ivanov** [comment]
Fixed #33.
